src/ServiceControl/Recoverability/Retrying/RetryProcessor.cs

```diff
@@ -195,6 +198,57 @@ async Task<int> Stage(RetryBatch stagingBatch, IAsyncDocumentSession session)
             return messages.Length;
         }
 
+        async Task TryStageMessage(FailedMessage message, string stagingId, FailedMessageRetry failedMessageRetry)
+        {
+            try
+            {
+                await StageMessage(message, stagingId).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                if (failedMessageRetry.StageAttempts < 5)
+                {
+                    await IncrementAttemptCounter(failedMessageRetry).ConfigureAwait(false);
+                    //We need to rethrow here to stop the current staging process
+                    throw new Exception($"Error while trying to stage failed message {message.UniqueMessageId}", e);
```

Comment:
Can this exception message and the one below be differentiated (so we can tell which one was thrown).
Should we also include the attempt count?
Out of interest, what happens to this exception? Is it logged? Is the inner exception logged correctly?

Reply:
At the moment this exception will cause the system to wait 30 seconds before trying again. As we're going to retry up to 5 times, a truly poison message that cannot be staged will halt the retries system for 3 minutes before being able to proceed.

Reply:
I tried to redesign this. Now throwing and catching a custom exception to trigger immediate retry of staging.